src/nomad_simulations/schema_packages/workflow/dft_plus_tb.py

```diff
@@ -0,0 +1,157 @@
+    )
+    tb_method_ref = Quantity(
+        type=Reference(TB),
+        description="""
+        Reference to the TB `ModelMethod` section in the TB task.
+        """,
+    )
+
+
+class DFTPlusTB(BeyondDFT):
+    """
+    A base section used to represent a DFT+TB calculation workflow. The `DFTPlusTB` workflow is composed of
+    two tasks: the initial DFT calculation + the final TB projection.
+
+    The section only needs to be populated with (everything else is handled by the `normalize` function):
+        i. The `tasks` as `TaskReference` sections, adding `task` to the specific archive.workflow2 sections.
+        ii. The `inputs` and `outputs` as `Link` sections pointing to the specific archives.
+
+    Note 1: the `inputs[0]` of the `DFTPlusTB` coincides with the `inputs[0]` of the DFT task (`ModelSystem` section).
+    Note 2: the `outputs[-1]` of the `DFTPlusTB` coincides with the `outputs[-1]` of the TB task (`Outputs` section).
+    Note 3: the `outputs[-1]` of the DFT task is used as `inputs[0]` of the TB task.
+
+    The archive.workflow2 section is:
+        - name = 'DFT+TB'
+        - method = DFTPlusTBMethod(
+            dft_method_ref=dft_archive.data.model_method[-1],
+            tb_method_ref=tb_archive.data.model_method[-1],
+        )
+        - inputs = [
+            Link(name='Input Model System', section=dft_archive.data.model_system[0]),
+        ]
+        - outputs = [
+            Link(name='Output TB Data', section=tb_archive.data.outputs[-1]),
+        ]
+        - tasks = [
+            TaskReference(
+                name='DFT SinglePoint Task',
+                task=dft_archive.workflow2
+                inputs=[
+                    Link(name='Input Model System', section=dft_archive.data.model_system[0]),
+                ],
+                outputs=[
+                    Link(name='Output DFT Data', section=dft_archive.data.outputs[-1]),
+                ]
+            ),
+            TaskReference(
+                name='TB SinglePoint Task',
+                task=tb_archive.workflow2,
+                inputs=[
+                    Link(name='Output DFT Data', section=dft_archive.data.outputs[-1]),
+                ],
+                outputs=[
+                    Link(name='Output tb Data', section=tb_archive.data.outputs[-1]),
+                ]
+            ),
+        ]
+    """
+
+    @check_n_tasks(n_tasks=2)
+    def link_task_inputs_outputs(
+        self, tasks: list[TaskReference], logger: 'BoundLogger'
+    ) -> None:
+        if not self.inputs or not self.outputs:
+            logger.warning(
+                'The `DFTPlusTB` workflow needs to have `inputs` and `outputs` defined in order to link with the `tasks`.'
+            )
+            return None
+
+        dft_task = tasks[0]
+        tb_task = tasks[1]
+
+        # Initial check
+        if not dft_task.m_xpath('task.outputs'):
+            return None
+
+        # Input of DFT Task is the ModelSystem
+        dft_task.inputs = [
```

Comment:
not sure if i understood this right but it should be the other way around right? i.e you assign self.in(out)puts from dft and tb task inputs and outputs

Reply:
After our discussion, this is what was confusing me with the need of defining inputs/outputs in a TaskReference.

I think I can improve this in a new schema as specified in https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/2143 with some minor tweeks